### ISSUE TITLE
[MAC] Addon immediately throws exception in Blender 4

### ISSUE THREAD

**Issue by hot-portafilter** [domlysz/BlenderGIS#851]
# **Blender and OS versions**

* Blender 4.0.2
* MacOS Sonoma 14.2.1

# **Describe the bug**

Blender UI locks up and ~/.bgis/bgis.log shows an error.

GIS &gt; Web geodata &gt; Basemap, Ok with defaults (Google &amp; Satellite)

In ~/.bgis/bgis.log:

DEBUG:BlenderGIS-228.core.basemaps.mapservice:726:77 tiles requested, 0 already in cache, 77 remains to download
DEBUG:BlenderGIS-228.core.basemaps.mapservice:580:http://mt0.google.com/vt/lyrs=s&x=0&y=0&z=0
ERROR:BlenderGIS-228.core.basemaps.mapservice:885:Corrupted tile on cache
...
  File "/Users/{user}/Library/Application Support/Blender/4.0/scripts/addons/BlenderGIS-228/core/lib/imageio/core/functions.py", line 107, in get_reader
    raise ValueError('Could not find a format to read the specified file '
ValueError: Could not find a format to read the specified file in mode 'i'

Full error log and tile file in attached zip. I tried deleting the cache but same error reoccurs systematically.
[fail.zip](https://github.com/domlysz/BlenderGIS/files/13811385/fail.zip)

**Comment by ssgraham767** [comment]
@hot-portafilter & @domlysz 

Its a bit clunky but I have managed to fix this (on OSX) with a couple changes. The issue is that there is no freeimage support for ARM architectures from the source provided in the FAQ. However, homebrew has a version that works. 

So...
1. Run `brew install freeimage`
2. `brew list freeimage` to locate the installed files. 
3. Copy the appropriate dylib into your freeimage folder. In my case, I put `libfreeimage.3.18.0.dylib` into /Users/<yourUserHere>/Library/Application Support/Blender/3.6/scripts/addons/BlenderGIS-master/core/lib/imageio/resources/freeimage/libfreeimage.3.18.0.dylib
4. Open /Users/<yourUserHere>/Library/Application Support/Blender/3.6/scripts/addons/BlenderGIS-master/core/lib/imageio/plugins/_freeimage.py in a code editor
5. Edit lines 31-38 to match the following:
```
FNAME_PER_PLATFORM = {
    'osx32': 'libfreeimage.3.18.0.dylib',  # universal library
    'osx64': 'libfreeimage.3.18.0.dylib',
    'win32': 'FreeImage-3.15.4-win32.dll',
    'win64': 'FreeImage-3.15.1-win64.dll',
    'linux32': 'libfreeimage-3.16.0-linux32.so',
    'linux64': 'libfreeimage-3.16.0-linux64.so',
}
```

You probably only need to alter the pointer for osx64. Make it match the name of your copied dylib file (in my case, `libfreeimage.3.18.0.dylib`).

For me, Blender's buttons don't work while navigating the map. So, use 'command'+'+' to zoom, and 'e' to select an area. After doing so, your buttons should work again—or at least they do for me. Now, hopefully, you can proceed.

Hope this helps!

**Comment by frankievx** [comment]
You're a legend! Thank you, this fixed my issue on my M2 Macbook Pro Sonoma 14.1.1! I installed using the branch on your fork

**Comment by domlysz** [comment]
fixed https://github.com/domlysz/BlenderGIS/releases/tag/2210
